### ISSUE TITLE
Can `FairRootManager::RegisterAny` consume `T*` instead of `T*&` as second argument?

### ISSUE THREAD

**Issue by dennisklein** [FairRootGroup/FairRoot#790]
Followup on one aspect identified in #789.

Disadvantages of `T*&`:
* If people want to register object that are referenced by a smart pointer, they are currently told to extract the raw pointer and extend its lifetime with a local static variable.
* The calling method will effectively only work once.
* You cannot pass rvalues.

**Comment by sawenzel** [comment]
I agree that there is room for improvement here and I appreciate the initiative. I remember it simply did not work with just `T*`. Some limitations come in my understanding from the way you need to pass addresses to the TBranch.
